LF-4562 remove unused animal sections

antsgar opened the pull request:
**Description**

- Removed Locations and Groups sections in Animals module. Animals now doesn't have subsections in the main menu.
- Removed group relations from animal and batch models, and removed code related to animals from animal/batch controllers and tests.
- Removed group routes from API.
- Removed group column and group filters from inventory UI, group name text field from animal creation flow, and all other group related code from the UI.
- Removed "Clone" and "Add to group" actions from action menu.

Jira link:
https://lite-farm.atlassian.net/browse/LF-4562

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

Duncan-Brain commented:
Wondering if we can comment out backend stuff too? It might be nice to prevent insertions on group table on all api accessible routes. Just so when we return to it we have a guaranteed clean db.

antsgar commented:
@Duncan-Brain yeah for sure, are you referring to the group related routes? I can comment out those.

Duncan-Brain commented:
Those yes, but it is also possible to add groups on add/edit animals/batches. I am not sure whether the `noInsert` option or removing model `relations`, or removing controller logic would be best.

Duncan-Brain commented:
@antsgar so far seeing 'groups' in the `withGraphFetched()` expression in taskController. That is the main error I see in the log. Will test and see if thats it.

antsgar commented on packages/webapp/src/containers/Animals/Inventory/index.tsx:
```diff
@@ -422,9 +403,7 @@ export default function AnimalInventory({
 
   const iconActions: iconAction[] = [
-    { label: t(`common:ADD_TO_GROUP`), iconName: 'ADD_ANIMAL', onClick: () => ({}) },
     { label: t(`common:CREATE_A_TASK`), iconName: 'TASK_CREATION', onClick: () => ({}) },
```
Kept this one because I added the feature on #3564

kathyavini reviewed (approved):
Has a merge conflict but it looks good to me!

Is there a difference between the tests that were skipped vs the ones deleted?

antsgar commented:
@kathyavini I deleted all the logic related to groups from the animal/batch controllers and models (I was initially going to comment it out but thought it might end up being too confusing for future readers), and so since that functionality no longer exists in the code I removed the associated tests as well. For groups, I kept the routes, controller and model and so I kept the tests. Not sure if this makes total sense though, I wasn't sure! I could also just skip the test that was entirely removed on the animal test file

kathyavini reviewed (approved):
> so since that functionality no longer exists in the code I removed the associated tests as well

That distinction makes sense to me!

SayakaOno reviewed (commented):
Just one thing to revert!
Could we keep the translations? I would feel bad for the translators if they had to translate the same strings again...

Duncan-Brain reviewed (dismissed):
Awesome thanks for doing this!

Edit: saw Sayaka is also reviewing haha ... undo my approval!

Question: should we immediately make a "revert" PR for future us?

Duncan-Brain commented on packages/webapp/src/containers/Animals/Inventory/useAnimalInventory.tsx:
```diff
@@ -43,17 +41,15 @@ import { useSelector } from 'react-redux';
 import { Location } from '../../../types';
 
-export type AnimalInventory = {
+export type AnimalInventoryItem = {
```
Thanks for improving this!

kathyavini commented:
> Question: should we immediately make a "revert" PR for future us?

@Duncan-Brain Aren't we doing a sensors release before we add anything to animals, though?

@antsgar you have another conflict, sorry. At this point it feels like every merge is a conflict in three other PRs 😅

Duncan-Brain commented:
@kathyavini yeah, just asking if we should make a PR -- not merge it. Probably answer is no haha

antsgar commented on packages/webapp/src/App.jsx:
```diff
@@ -24,12 +24,12 @@ import { NotistackSnackbar } from './containers/Snackbar/NotistackSnackbar';
   const [isCompactSideMenu, setIsCompactSideMenu] = useState(false);
   const [isFeedbackSurveyOpen, setFeedbackSurveyOpen] = useState(false);
-  const FULL_WIDTH_ROUTES = ['/map', ANIMALS_INVENTORY_URL, ANIMALS_URL];
+  const FULL_WIDTH_ROUTES = ['/map', ANIMALS_INVENTORY_URL];
```
Good call!!

antsgar commented:
@Duncan-Brain it'd be pretty easy to create one with a revert commit from the merge commit once this is merged, so I think we could do that when the time comes!

@SayakaOno I readded the translation strings, good call ☺️ this is ready for re-reviews I think
